sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -675,15 +676,19 @@ object JdbcUtils extends Logging {
             }
             i = i + 1
           }
-          stmt.addBatch()
-          rowCount += 1
-          totalRowCount += 1
-          if (rowCount % batchSize == 0) {
-            stmt.executeBatch()
-            rowCount = 0
+          if (supportsBatchUpdates) {
```

Comment:
Could you pull this check out from the loop? I think we don't need to check it in every iterations.